controllers/openstackplaybookgenerator_controller.go

```diff
@@ -111,6 +111,85 @@ func (r *OpenStackPlaybookGeneratorReconciler) Reconcile(ctx context.Context, re
 	templateParameters := make(map[string]interface{})
 	cmLabels := common.GetLabels(instance, openstackplaybookgenerator.AppLabel, map[string]string{})
 
+	// check if heat-env-config (customizations provided by administrator) exist
+	// if it does not exist, requeue
+	tripleoCustomDeployCM, _, err := common.GetConfigMapAndHashWithName(r, instance.Spec.HeatEnvConfigMap, instance.Namespace)
+	if err != nil {
+		if k8s_errors.IsNotFound(err) {
+			msg := fmt.Sprintf("The ConfigMap %s doesn't yet exist. Requeing...", instance.Spec.HeatEnvConfigMap)
+			r.Log.Info(msg)
+			err = r.setCurrentState(instance, ospdirectorv1beta1.PlaybookGeneratorWaiting, msg)
+			return ctrl.Result{RequeueAfter: time.Second * 10}, err
+		}
+		// Ignore any potential error from "setCurrentState"; focus on previous error
+		// NOTE: This pattern will be repeated a lot below, but the comment will not be copied
+		_ = r.setCurrentState(instance, ospdirectorv1beta1.PlaybookGeneratorError, err.Error())
+		return ctrl.Result{}, err
+	}
+
+	tripleoCustomDeployFiles := tripleoCustomDeployCM.Data
+	templateParameters["TripleoCustomDeployFiles"] = tripleoCustomDeployFiles
+
+	// Now read the tripleo-deploy-config and the fencing-config CMs for use in the PlaybookGenerator
+	tripleoDeployCM, _, err := common.GetConfigMapAndHashWithName(r, "tripleo-deploy-config", instance.Namespace)
```

Comment:
just realized we do not have a const for tripleo-deploy-config and use it now in couple of places. Might be good we change that in a later change.

Reply:
depending on how we implement DCN that can't be a const... so lets wait for the work on that feature